lib/jekyll/commands/serve.rb

```diff
@@ -218,6 +218,25 @@ def start_up_webrick(opts, destination)
           Jekyll.logger.info "Server address:", server_address(@server, opts)
           launch_browser @server, opts if opts["open_url"]
           boot_or_detach @server, opts
+        rescue StandardError => e
+          server_error_message(e)
+        end
+
+        def server_error_message(exception)
+          server_msg = "Could not start server:"
+
+          case exception
+          when Errno::EADDRINUSE
+            Jekyll.logger.error server_msg,
+                                "Port is already in use."
+          when Errno::EACCES
+            Jekyll.logger.error server_msg,
+                                "Access denied for address or port."
+          else
+            Jekyll.logger.error server_msg, exception.message.to_s
+          end
+
+          raise exception
         end
```

Comment:
You may refactor this method to be more compact yet more informative.. *For example*,
```ruby
def server_error_message(exception, address, port)
  message = case exception
            when Errno::EADDRINUSE
              "Port '#{port}' is already in use."
            when Errno::EACCES
              "Access denied to address '#{address}'."
            else
              exception.to_s  # NOTE: Prefer `e.to_s` over `e.message.to_s`
            end

  # NOTE:  The first argument to Jekyll's logger is best rendered at 20 chars or less.
  Jekyll.logger.error "Server Error:", message
  raise exception
end
```

Reply:
Thank you for the suggestions.

It appears that `opts["address"]` may be empty at the time of the error, so I have only added `port` information.

Note that `server_address` pulls the address/port information from the WEBrick instance, which is also not available at the time of the exception.

Reply:
You could stash the sever address in a local variable before hand and pass it down on error, no?
```ruby
def start_up_webrick(opts, destination)
  ...
  address = server_address(@server, opts)
  Jekyll.logger.info "Server address:", address
  ...
rescue StandardError => e
  server_error_message(e, opts["port"], address)
end
```

Reply:
Oh I don't think I grabbed the right hash key.

However, I don't think the address is that important for the error message, honestly. I don't know if there's a case where you can't bind to an _address_ due to a permissions issue, it's almost always port related. I can adjust but I think it's less confusing to just reference the port number.

Reply:
okay.. I suggested this because we allow the *host* and *scheme* aspects of the address to be configured as well.

Reply:
good

- [ ]